integrations/openai/src/databricks_openai/vector_search_retriever_tool.py

```diff
@@ -125,6 +125,15 @@ def get_tool_name():
             description=self.tool_description
             or self._get_default_tool_description(self._index_details),
         )
+
+        try:
+            from databricks.sdk import WorkspaceClient
+
+            WorkspaceClient().serving_endpoints.get(self.embedding_model_name)
+            self.resources = self._get_resources(self.index_name, self.embedding_model_name)
```

Comment:
Here for resources for embedding, we are checking if the embedding model is a model serving endpoint that we can grant access for. So doing a quick check here. If the resource doesn't exist we fallback to not returning back resources for endpoint here.